dbt_platform_helper/domain/conduit.py

```diff
@@ -94,8 +95,16 @@ def start_task(self, data_context: dict):
                 },
             ]
 
+        cluster_name = (
+            f"{self.application.name}-{self.env}-cluster"
+            if self.ecs_provider.get_cluster_arn_by_name(
```

Comment:
instead of calling the ecs provider again you could check the arn in the data dict as the arn for a cluster contains the name